src/Appwrite/Auth/OAuth2.php

```diff
@@ -62,9 +62,16 @@ abstract public function getLoginURL():string;
     /**
      * @param string $code
      *
-     * @return string
+     * @return array
+     */
+    abstract protected function getTokens(string $code):array;
+
+    /**
+     * @param string $refreshToken
+     *
+     * @return array
      */
-    abstract public function getAccessToken(string $code):string;
+    abstract public function refreshTokens(string $refreshToken):array;
```

Comment:
`refreshTokens` or `refreshToken`?

Reply:
Tokens. Auth0 says `With refresh token rotation enabled, every time an application exchanges a refresh token to get a new access token, a new refresh token is also returned `.

This means refresh token can also change when the access token is renewed, so one or two tokens could be changing, depending on specific OAuth provider.